File: java solutions/1. Two Sum.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {
      Map<Integer, Integer> values = new HashMap<>();
      for(int i = 0 ; i<nums.length; i++){
        int find = target - nums[i];
        if(values.containsKey(find)){
          return new int[]{values.get(find), i};
        }
        values.put(nums[i], i);
      }
      return new int[]{};
        
    }
}

File: python solutions/1. Two Sum.py
#brute force solution
#Time complexity: O(n^2)
#two nested loops are used to iterate through the list and find the sum of two numbers equal to the target

from typing import List
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
      for i in range(len(nums)):
        for j in range(i+1, len(nums)):
          if(nums[i]+nums[j] == target):
            return [i,j]
            break
      

#optimized solution
#Time complexity: O(n)
#map is used to store the difference between the target and the current element